dask_geopandas/core.py

```diff
@@ -0,0 +1,330 @@
+        setattr(cls, name, property(fget=prop, doc=doc))
+
+    @classmethod
+    def _bind_elemwise_comparison_method(
+        cls, name, comparison, original, *args, **kwargs
+    ):
+        """ bind comparison method like GeoSeries.contains to this class """
+
+        def meth(self, other, *args, **kwargs):
+            return elemwise(comparison, self, other, *args, **kwargs)
+
+        meth.__name__ = name
+        setattr(cls, name, derived_from(original)(meth))
+
+    @classmethod
+    def _bind_elemwise_operator_method(cls, name, op, original, *args, **kwargs):
+        """ bind operator method like GeoSeries.distance to this class """
+        # name must be explicitly passed for div method whose name is truediv
+        def meth(self, other, *args, **kwargs):
+            meta = _emulate(op, self, other)
+            return map_partitions(
+                op, self, other, meta=meta, enforce_metadata=False, *args, **kwargs
+            )
+
+        meth.__name__ = name
+        setattr(cls, name, derived_from(original)(meth))
+
+    @property
+    def crs(self):
+        """
+        The Coordinate Reference System (CRS) represented as a ``pyproj.CRS``
+        object.
+
+        Returns None if the CRS is not set, and to set the value it
+        :getter: Returns a ``pyproj.CRS`` or None. When setting, the value
+        can be anything accepted by :meth:`pyproj.CRS.from_user_input`,
+        such as an authority string (eg "EPSG:4326") or a WKT string.
+        """
+        return self._meta.crs
+
+    @crs.setter
+    def crs(self, value):
+        """Sets the value of the crs"""
+        new = self.set_crs(value)
+        self._meta = new._meta
+        self.dask = new.dask
+
+    def set_crs(self, value):
+        """Set the value of the crs on a new object"""
+
+        def set_crs(df, crs):
+            df = df.copy(deep=False)
+            df.crs = crs
+            return df
+
+        return self.map_partitions(set_crs, value, enforce_metadata=False)
+
+    @property
+    @derived_from(geopandas.base.GeoPandasBase)
+    def total_bounds(self):
+        def agg(concatted):
+            return np.array(
+                (
+                    concatted[0::4].min(),  # minx
+                    concatted[1::4].min(),  # miny
+                    concatted[2::4].max(),  # maxx
+                    concatted[3::4].max(),  # maxy
+                )
+            )
+
+        return self.reduction(
+            lambda x: getattr(x, "total_bounds"),
+            token=self._name + "-total_bounds",
+            meta=self._meta.total_bounds,
+            aggregate=agg,
+        )
+
+    @property
+    def sindex(self):
+        """Need to figure out how to concatenate spatial indexes"""
+        raise NotImplementedError
+
+    @property
+    @derived_from(geopandas.base.GeoPandasBase)
+    def unary_union(self):
+        attr = "unary_union"
+        meta = GeometryCollection()
```

Comment:
I think BaseGeometry is better here, since eg unary union of points gives MultiPoint, which is not a GeometryCollection subclass

(not sure if that actually matters though)

Reply:
ok!